### ISSUE TITLE
Should np.asarray(df) have the same behavior as df.to_numpy?

### ISSUE THREAD

**Issue by thomasjpfan** [pola-rs/polars#7961]
### Polars version checks

- [X] I have checked that this issue has not already been reported.
- [X] I have confirmed this bug exists on the [latest version](https://pypi.org/project/polars/) of Polars.

### Issue description

Currently, `np.asarray(df)` returns the transposed of `df.to_numpy()`. Should `np.asarray(df)` return the same ndarray?

### Reproducible example

```python
import polars as pl
import numpy as np

df = pl.DataFrame({"a": [1, 2, 3], "b": [3, 4, 5]})

print(df.to_numpy())
# [[1 3]
# [2 4]
# [3 5]]

print(np.asarray(df))
# [[1 2 3]
# [3 4 5]]
```

### Expected behavior

I expect `np.asarray(df)` to give the same results as `df.to_numpy()`.

### Installed versions

<details>

```
---Version info---
Polars: 0.16.18
Index type: UInt32
Platform: macOS-13.2.1-arm64-arm-64bit
Python: 3.10.10 | packaged by conda-forge | (main, Mar 24 2023, 20:12:31) [Clang 14.0.6 ]
---Optional dependencies---
numpy: 1.23.1
pandas: 1.4.3
pyarrow: 11.0.0
connectorx: <not installed>
deltalake: <not installed>
fsspec: <not installed>
matplotlib: <not installed>
xlsx2csv: <not installed>
xlsxwriter: <not installed>
```

</details>

**Comment by StevenLi-DS** [comment]
This seems like the default behavior in pandas as well https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.to_numpy.html

**Comment by alexander-beedie** [comment]
Closed by #7979.
